Src/ILGPU/Backends/OpenCL/CLCodeGenerator.Emitter.cs

```diff
@@ -395,7 +395,10 @@ public void AppendConstant(float value)
             {
                 AppendArgument();
                 stringBuilder.Append(value);
-                stringBuilder.Append('f');
+                if (value % 1 == 0)
```

Comment:
Please change the constants `1` and `0` to `1.0f` and `0.0f`. After that this PR should be ready to go 👍